Setup Travis and basic file structure

rwblickhan opened the pull request:
I created a `main` function as well as a basic directory structure, and also set up Travis so that it can build on every pull request (this will however need to be activated on [travis-ci.org](travis-ci.org)). Also I set the C++ standard to C++14 for the time being; it doesn't change much from C++11 (i.e. modern C++) but since we don't really have to worry about backwards compatibility and cross-platform and so on, we might as well use the most recent version.

rwblickhan committed:
Create source and include directories and program entrance file

rwblickhan committed:
Add CMake config file

rwblickhan committed:
Activate travis

bobheadxi commented on CMakeLists.txt:
```diff
@@ -0,0 +1,12 @@
+
+set(CMAKE_CXX_STANDARD 14)
+
+include_directories(${PROJECT_SOURCE_DIR}/src/include)
```
Shouldn't we `.gitignore` the cmake output?

bwdmonkey replied:
I'd include the executables after `make` is ran as well.

rwblickhan replied:
Yeah, we should. I was also going to include IDE files but wanted by to ask around what IDEs y’all were planning to use first.

bobheadxi replied:
VScode is life sir

bwdmonkey replied:
VSCode

rwblickhan replied:
Do you guys happen to know what IDE files VScode uses?

bobheadxi replied:
vscode config is in `.vscode/settings.json` 

Standard setup procedure: https://code.visualstudio.com/docs/languages/cpp

bobheadxi reviewed (approved):
Looks great! 🔥 Only thing that might need changing is a `.gitignore`

```cpp
#include <stdio.h>

int main() {
    printf("Hello world!\n");
    return 0;
}
```

```bash
bobbook:durian robertlin$ cmake CMakeLists.txt
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/robertlin/Projects/durian
bobbook:durian robertlin$ make
Scanning dependencies of target durian
[ 50%] Building CXX object CMakeFiles/durian.dir/src/main.cpp.o
[100%] Linking CXX executable durian
[100%] Built target durian
bobbook:durian robertlin$ ./durian
Hello world!
```

bwdmonkey reviewed (commented):
LGTM. I assume we put the header files inside `src/include/` and the `.cpp` files in the `src/`?

rwblickhan committed:
Add .gitignore

rwblickhan commented:
Yep, `.h` goes into `src/include/` and `.cpp` goes into `src/`, or at least that's the intention. Presumably we would also add subdirectories of `src/` for each major component (i.e. `src/lexer/`, `src/parser/`, etc), and *maybe* mirror that in `src/include/`.

But that's just one way of doing things (albeit one I prefer 😛) and we could discuss other ways of structuring the project.

rwblickhan committed:
Add 'building' section to README

rwblickhan commented:
I've added the .gitignore. I also added a section on building to the README which explains how to get it to build if you want to do it from the command line (which is useful for working with Travis, for instance).
